clang/labs/lab1/petrov_maksim/WarningDeprecatedPlugin.cpp

```diff
@@ -0,0 +1,65 @@
+#include "clang/AST/ASTConsumer.h"
+#include "clang/AST/RecursiveASTVisitor.h"
+#include "clang/Frontend/CompilerInstance.h"
+#include "clang/Frontend/FrontendPluginRegistry.h"
+
+using namespace clang;
+
+class WarningDepricatedVisitor
+    : public RecursiveASTVisitor<WarningDepricatedVisitor> {
+private:
+  ASTContext *Context;
+
+public:
+  explicit WarningDepricatedVisitor(ASTContext *Context) {this->Context = Context;}
```

Comment:
Could you please pull last changes from main branch and fix clang-format and clang-tidy tests?